java/vector/src/main/java/org/apache/arrow/vector/ipc/ArrowStreamWriter.java

```diff
@@ -52,4 +54,9 @@ protected void endInternal(WriteChannel out,
                              List<ArrowBlock> records) throws IOException {
     out.writeIntLittleEndian(0);
   }
+
+  @Override
+  protected void writeRecordBatch(ArrowRecordBatch batch) throws IOException {
```

Comment:
I am not sure how this method helps in saving memory overhead as mentioned in PR description. Can you please elaborate on the usage of this method?